warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java

```diff
@@ -326,68 +330,67 @@ public DatawaveFieldIndexCachingIteratorJexl() {
-        this(builder.fieldName, builder.fieldValue, builder.timeFilter, builder.datatypeFilter, builder.negated, builder.scanThreshold, builder.scanTimeout,
-                        builder.hdfsBackedSetBufferSize, builder.maxRangeSplit, builder.maxOpenFiles, builder.fs, builder.uniqueDir, builder.queryLock,
-                        builder.allowDirReuse, builder.returnKeyType, builder.sortedUIDs, builder.compositeMetadata, builder.compositeSeekThreshold,
-                        builder.typeMetadata, builder.env);
-    }
-    
-    @SuppressWarnings("hiding")
-    private DatawaveFieldIndexCachingIteratorJexl(Text fieldName, Text fieldValue, TimeFilter timeFilter, Predicate<Key> datatypeFilter, boolean neg,
-                    long scanThreshold, long scanTimeout, int bufferSize, int maxRangeSplit, int maxOpenFiles, FileSystem fs, Path uniqueDir,
-                    QueryLock queryLock, boolean allowDirReuse, PartialKey returnKeyType, boolean sortedUIDs, CompositeMetadata compositeMetadata,
-                    int compositeSeekThreshold, TypeMetadata typeMetadata, IteratorEnvironment env) {
-        if (fieldName.toString().startsWith("fi" + NULL_BYTE)) {
-            this.fieldName = new Text(fieldName.toString().substring(3));
-            this.fiName = fieldName;
+        if (builder.fieldName.toString().startsWith("fi" + NULL_BYTE)) {
+            this.fieldName = new Text(builder.fieldName.toString().substring(3));
+            this.fiName = builder.fieldName;
         } else {
-            this.fieldName = fieldName;
-            this.fiName = new Text("fi" + NULL_BYTE + fieldName);
+            this.fieldName = builder.fieldName;
+            this.fiName = new Text("fi" + NULL_BYTE + builder.fieldName);
         }
         log.trace("fName : " + fiName.toString().replaceAll(NULL_BYTE, "%00"));
-        this.fieldValue = fieldValue;
-        this.negated = neg;
-        this.returnKeyType = returnKeyType;
-        this.timeFilter = timeFilter;
-        this.datatypeFilter = datatypeFilter;
-        
-        this.fs = fs;
-        this.queryLock = queryLock;
-        this.uniqueDir = uniqueDir;
-        this.allowDirReuse = allowDirReuse;
-        this.scanThreshold = scanThreshold;
-        this.scanTimeout = scanTimeout;
-        this.hdfsBackedSetBufferSize = bufferSize;
-        this.maxOpenFiles = maxOpenFiles;
-        this.maxRangeSplit = maxRangeSplit;
-        
-        this.sortedUIDs = sortedUIDs;
+        this.fieldValue = builder.fieldValue;
+        this.negated = builder.negated;
+        this.returnKeyType = builder.returnKeyType;
+        this.timeFilter = builder.timeFilter;
+        this.datatypeFilter = builder.datatypeFilter;
+        
+        this.ivaratorCacheDirs = builder.ivaratorCacheDirs;
+        
+        if (ivaratorCacheDirs.size() > 0) {
+            this.controlFs = ivaratorCacheDirs.get(0).getFs();
+            this.controlDir = new Path(ivaratorCacheDirs.get(0).getPathURI());
```

Comment:
is there any reason not to make these a separate config instead of by convention using the first?

Reply:
It's a little bit trickier than that actually.  Prior to the ivarator cache directories being passed to the query iterator, their order is randomized (respecting priority of course).  So, if you have 5 ivarator cache dirs with priority 0, they will be shuffled, and the first one will be used as the 'controlDir'.  This is intended to spread out the load among the various ivarator cache directories.  Hope that clears things up.

Reply:
done